bundles/framework/myplaces3/ButtonHandler.js

```diff
@@ -0,0 +1,417 @@
+
+            var addMeasureTool = function(tool){
+                var drawOpt = {'showMeasureOnMap':true, 'allowMultipleDrawing':false};
+                if (tool === 'line') {
+                    measureTool = jQuery.extend(true, {}, me.buttons[tool]);
+                    measureTool.callback = function () {
+                        me.sendDrawRequest({
+                            drawMode: 'line',
+                            shape: "LineString",
+                            drawOptions: drawOpt
+                        });
+                    };
+                    measureTool.iconCls = 'tool-measure-line';
+                    measureTool.tooltip = me.loc('tools.measureline.tooltip');
+                    sandbox.request(me, reqBuilder(tool, me.measureButtonGroup, measureTool));
+                }
+                if (tool === 'area') {
+                    measureTool = jQuery.extend(true, {}, me.buttons[tool]);
+                    measureTool.callback = function () {
+                        me.sendDrawRequest({
+                            drawMode: 'area',
+                            shape: "Polygon",
+                            drawOptions: drawOpt
+                        });
+                    };
+                    measureTool.iconCls = 'tool-measure-area';
+                    measureTool.tooltip = me.loc('tools.measurearea.tooltip');
+                    sandbox.request(me, reqBuilder(tool, me.measureButtonGroup, measureTool));
+                }
+            }; */
+
+            for (tool in this.buttons) {
+                if (!this.buttons.hasOwnProperty(tool)) {
+                    continue;
+                }
+                sandbox.request(this, reqBuilder(tool, this.buttonGroup, this.buttons[tool]));
+                
+                // for logged-in-user: add line & area measurement tools if configured
+                // basic measurement tools should be configured off so these can be used
+                // as replacements - this way measurement can be saved as myplaces
+                // TODO: this is not the way to do this: instead allow request to be used for
+                // saving a place and make the basic measurement tools use it when available
+                // TODO uncomment this if you need to add AdditionalMeasureTools
+                /*
+                if (Oskari.user().isLoggedIn() && addAdditionalMeasureTools) {
+                    if (tool === 'line') {
+                        addMeasureTool(tool);
+                    }
+                    if (tool === 'area') {
+                        addMeasureTool(tool);
+                    }
+                }*/
+            }
+
+            var user = Oskari.user();
+            if (!user.isLoggedIn()) {
+                // disable toolbar buttons for guests
+                this.disableButtons();
+            } else {
+                // logged in user -> listen to events as normal
+                for (p in me.eventHandlers) {
```

Comment:
You can do this in a more compact way with:

    Object.keys(this.eventHandlers).forEach(function (eventName) {
        sandbox.registerForEventByName(this, eventName);
    }, this);

But it doesn't really matter which way you do it.